packages/core/src/telemetry/metrics.ts

```diff
@@ -124,84 +386,20 @@ export function initializeMetrics(config: Config): void {
   if (!meter) return;
 
   // Initialize core metrics
-  toolCallCounter = meter.createCounter(METRIC_TOOL_CALL_COUNT, {
-    description: 'Counts tool calls, tagged by function name and success.',
-    valueType: ValueType.INT,
-  });
-  toolCallLatencyHistogram = meter.createHistogram(METRIC_TOOL_CALL_LATENCY, {
-    description: 'Latency of tool calls in milliseconds.',
-    unit: 'ms',
-    valueType: ValueType.INT,
-  });
-  apiRequestCounter = meter.createCounter(METRIC_API_REQUEST_COUNT, {
-    description: 'Counts API requests, tagged by model and status.',
-    valueType: ValueType.INT,
-  });
-  apiRequestLatencyHistogram = meter.createHistogram(
-    METRIC_API_REQUEST_LATENCY,
-    {
-      description: 'Latency of API requests in milliseconds.',
-      unit: 'ms',
-      valueType: ValueType.INT,
+  Object.entries(COUNTER_DEFINITIONS).forEach(
```

Comment:
Prefer classical `for... of... ` loop.

The .foreach is kind of discouraged because all of the other methods on Array.prototype return a new array (e.g. `filter`, `reduce`, etc, etc).  Foreach is the one weird child that returns `void`.  For that reason, I almost always prefer `for ... of...` loops as it makes my intent very explicit.